2996_reflection/reflection.md

```diff
@@ -5339,18 +5390,76 @@ as specified in [dcl.type.simple]{.sref}. The guides of `A` are the set of funct
+[The changes to paragraph 2.3 (except for the wording related to `$splice-expression$s`) are a part of the resolution to CWG2701. These changes render [over.match.best.general]/4 redundant, hence the relocation of its associated example to this section.]{.ednote}
+
+Specify rules for overload sets denoted by `$splice-expression$`s in paragraph 2, make drive-by fixes to help clear up the situation more generally, and move the example that formerly followed [over.match.best.general]/4 to follow after paragraph 2 (with a new example covering `$splice-expression$`s).
 
 ::: std
 [2]{.pnum} First, to be a viable function, a candidate function shall have enough parameters to agree in number with the arguments in the list.
 
 - [#.#]{.pnum} If there are `$m$` arguments in the lists, all candidate functions having exactly `$m$` parameters are viable.
 - [#.#]{.pnum} A candidate function having fewer than `$m$` parameters is viable only if it has an ellipsis in its parameter list ([dcl.fct]). For the purposes of overload resolution, any argument for which there is no corresponding parameter is considered to "match the ellipsis" ([over.ics.ellipsis]).
-- [#.#]{.pnum} A candidate function having more than `$m$` parameters is viable only if
+- [#.#]{.pnum} A candidate function having more than `$m$` parameters is viable only if [there is a declaration `$D$` considered by the overload resolution such that for each]{.addu} [all]{.rm} parameter[s]{.rm} following the `$m$@^th^@` [have default arguments]{.rm}[, a reachable declaration whose host scope is the same as `$D$` specifies a default argument for that parameter]{.addu} ([dcl.fct.default]). [If the candidate function is selected as the best viable function, no other host scope of a declaration considered by overload resolution shall be the host scope of a declaration that specifies a default argument for the (`$m$`+1)^_st_^ parameter; the host scope of `$D$` shall not be a block scope if the declarations considered by overload resolution were denoted by a `$splice-expression$` ([expr.prim.splice]). The default arguments used in a call to the selected function are the default arguments introduced in the host scope of `$D$`.]{.addu} For the purposes of overload resolution, the parameter list is truncated on the right, so that there are exactly `$m$` parameters.
 
-  - [#.#.#]{.pnum} all parameters following the `$m$`^th^ have default arguments ([dcl.fct.default]) [and]{.addu}
-  - [#.#.#]{.pnum} [the set of candidate functions was not denoted by a `$splice-expression$` ([expr.prim.splice]).]{.addu}
+::: addu
+::: example
+```cpp
+namespace A {
+  extern "C" void f(int = 5);
+}
+namespace B {
+  extern "C" void f(int = 5);
+}
+
+void splice() {
+  [:^^A::f:]();  // error: found default argument twice
+}
+
+using A::f;
+using B::f;
+
+void use() {
+  f(3);          // OK, default argument was not used for viability
+  f();           // error: found default argument twice
```

Comment:
Can we add a normal example showing splicing working too?

Reply:
Done